wkcuber/wkcuber/cubing.py

```diff
@@ -299,18 +293,24 @@ def cubing(
         batch_size = BLOCK_LEN
 
     target_mag = Mag(target_mag_str)
-    target_wkw_info = WkwDatasetInfo(
-        target_path,
+
+    target_ds = Dataset.get_or_create(target_path, scale=scale)
+    target_layer = target_ds.get_or_add_layer(
         layer_name,
+        LayerCategories.COLOR_TYPE,
```

Comment:
Is this always a color layer?

Reply:
I would assume that it can also be segmentation, there should have been some heuristic (or user parameter) in place deciding about that. I can’t find that currently, though. Can you comment on that @normanrz or @youri-k ?

Reply:
I think there was a simple heuristic `segmentation if folder_name == "segmentation" else color`, that was implemented in the metadata inferral

Reply:
I have now modified the code to create a segmentation layer if the name is `"segmentation"`. However, this also requires the `largest_segment_id`. Currently I have set it simply to `-1`. Should I compute the largest_segment_id at the end, even though this might take a while for very large datasets?

Reply:
I don't know if `-1` is a valid value. I would default to `0` instead.
The cubing tool is reading all data anyway, would it be possible to accumulate the max value or is that too complex to  implement?

Reply:
Yes, that makes sense. I have modified `wait_and_ensure_success` to return a value for each cube (in this case the largest_segment_id).